codegen/masm/src/opt/operands/solver.rs

```diff
@@ -147,6 +157,7 @@ impl OperandMovementConstraintSolver {
                 self.tactics.push(Box::new(Linear));
                 self.tactics.push(Box::new(CopyAll));
             }
+            self.tactics.push(Box::new(TwoArgs));
```

Comment:
Consider adding a debug_assert before running the solver to ensure `TwoArgs` is running first. Some red light if we mess with the order in the future.